File: packages/agent-core/src/approval.ts
import type { ApprovalId, ToolCallId, RunId } from "./ids.js";

export type AgentApprovalDecision = "approved" | "denied" | "expired";

export type AgentApproval = {
  id: ApprovalId;
  toolCallId: ToolCallId;
  runId: RunId;
  reason: string;
  riskLevel: "safe" | "read" | "write" | "dangerous";
  redactedInput?: unknown;
  status: "pending" | "resolved" | "expired";
  decision?: AgentApprovalDecision;
  decidedBy?: string;
  decidedAt?: string;
  createdAt: string;
};

File: backend/src/aithru_agent/domain/approval.py
from enum import StrEnum

from .base import AithruBaseModel


class AgentApprovalDecision(StrEnum):
    APPROVED = "approved"
    REJECTED = "rejected"
    EXPIRED = "expired"


class AgentApprovalStatus(StrEnum):
    PENDING = "pending"
    RESOLVED = "resolved"
    EXPIRED = "expired"


class AgentApproval(AithruBaseModel):
    id: str
    run_id: str
    tool_call_id: str
    tool_name: str
    tool_input: dict | None = None
    status: AgentApprovalStatus
    decision: AgentApprovalDecision | None = None
    comment: str | None = None
    created_at: str
    resolved_at: str | None = None
